libnopegl/src/node_hblur.c

```diff
@@ -0,0 +1,674 @@
+        return ret;
+
+    if ((ret = setup_dummy_map(node)) < 0 ||
+        (ret = setup_pass1_pipeline(node)) < 0 ||
+        (ret = setup_pass2_pipeline(node)) < 0)
+        return ret;
+
+    return 0;
+}
+
+static int resize(struct ngl_node *node)
+{
+    int ret = 0;
+    struct ngl_ctx *ctx = node->ctx;
+    struct hblur_priv *s = node->priv_data;
+    struct hblur_opts *o = node->opts;
+
+    ngli_node_draw(o->source);
+    if (o->map)
+        ngli_node_draw(o->map);
+
+    struct texture_priv *src_priv = o->source->priv_data;
+    const int32_t width = src_priv->image.params.width;
+    const int32_t height = src_priv->image.params.height;
+    if (s->width == width && s->height == height)
+        return 0;
+
+    struct texture *dst = NULL;
+    struct texture *tex0 = ngli_texture_create(ctx->gpu_ctx);
+    struct texture *tex1 = ngli_texture_create(ctx->gpu_ctx);
+    struct rtt_ctx *pass1_rtt_ctx = ngli_rtt_create(ctx);
+    struct rtt_ctx *pass2_rtt_ctx = ngli_rtt_create(ctx);
+    if (!tex0 || !tex1 || !pass1_rtt_ctx || !pass2_rtt_ctx) {
+        ret = NGL_ERROR_MEMORY;
+        goto fail;
+    }
+
+    struct texture_params texture_params = {
+        .type          = NGLI_TEXTURE_TYPE_2D,
+        .format        = s->prefered_format,
+        .width         = width,
+        .height        = height,
+        .min_filter    = NGLI_FILTER_LINEAR,
+        .mag_filter    = NGLI_FILTER_LINEAR,
+        .wrap_s        = NGLI_WRAP_CLAMP_TO_EDGE,
+        .wrap_t        = NGLI_WRAP_CLAMP_TO_EDGE,
+        .usage         = NGLI_TEXTURE_USAGE_COLOR_ATTACHMENT_BIT |
+                         NGLI_TEXTURE_USAGE_SAMPLED_BIT,
+    };
+
+    ret = ngli_texture_init(tex0, &texture_params);
+    if (ret < 0)
+        goto fail;
+
+    ret = ngli_texture_init(tex1, &texture_params);
+    if (ret < 0)
+        goto fail;
+
+    struct rtt_params pass1_rtt_params = {
+        .width = width,
+        .height = height,
+        .nb_colors = 2,
+        .colors[0] = {
+            .attachment = tex0,
+            .store_op = NGLI_STORE_OP_STORE
+        },
+        .colors[1] = {
+            .attachment = tex1,
+            .store_op = NGLI_STORE_OP_STORE
+        },
+    };
+
+    ret = ngli_rtt_init(pass1_rtt_ctx, &pass1_rtt_params);
+    if (ret < 0)
+        goto fail;
+
+    /* Assert that the destination texture format does not change */
+    struct texture_priv *dst_priv = o->destination->priv_data;
+    ngli_assert(dst_priv->params.format == s->pass2.layout.colors[0].format);
+
+    dst = dst_priv->texture;
+    if (s->dst_is_resizeable) {
+        dst = ngli_texture_create(ctx->gpu_ctx);
+        if (!dst) {
+            ret = NGL_ERROR_MEMORY;
+            goto fail;
+        }
+
+        struct texture_params params = dst_priv->params;
+        params.width = width;
+        params.height = height;
+        ret = ngli_texture_init(dst, &params);
+        if (ret < 0)
+            goto fail;
+    }
+
+    if (s->dst_is_resizeable) {
```

Comment:
this is the same condition as above

Reply:
Moved the chunk below (where it belongs).

Reply:
Done.